docs/async-await.md

```diff
@@ -13,36 +12,35 @@ async function foo() {
         console.log('Error: ', err.message);
     }
 }
-```
 
-When the promise settles execution continues,
-* if it was fulfilled then await will return the value,
-* if it's rejected an error will be thrown synchronously which we can catch.
-
-This suddenly (and magically) makes asynchronous programming as easy as synchronous programming.  Three things are needed for this though experiment are.
+```
+Verilen söz bir sonuca ulaştığında program çalışmaya devam eder,
+* Eğer verilen söz yerine getirildiyse `await` verilen sözün sonucunu dönecektir,
+* Eğer verilen söz yerine getirilmediyse (reddedildiyse), çalışma zamanında yakalayabileceğimiz bir hata senkron olarak fırlatılacaktır  
 
-* Ability to *pause function* execution.
-* Ability to *put a value inside* the function.
-* Ability to *throw an exception inside* the function.
+Bu yaklaşım asenkron programlamayı bir anda senkron programlama gibi kolay bir hale getiriyor.Bu deney için üç bileşen gereklidir.  
+* *Fonksiyonları duraksatabilme* kabiliyeti.
+* Fonksiyonun içerisine *bir değer koyabilme* kabiliyeti.
+* Fonksiyonun içerisinden *bir istisna fırlatabilme* kabiliyeti.
 
-This is exactly what generators allowed us to do! The thought experiment *is actually real* and is the `async`/`await` implementation in TypeScript / JavaScript. Under the covers it just uses generators.
+İşte bu saydıklarımız tam olarak *oluşturucular*ın yapmamıza olanak tanıdığı şey. Aslında farzedilen deney bir kurgu değil gerçeğin ta kendisidir ve  Javascript / Typescript dünyasında `asenkron`/`bekle` implementasyonu olarak ifade edilir. Perdenin arkasında oluşturucuları kullanır.
```

Comment:
farzedilen -> farz edilen 
Öneri olarak: implementasyon için de uygulama / uygulanması kullanılabilir